hrm-service/src/permissions/middlewares.js

```diff
@@ -1,9 +1,8 @@
 const Actions = require('./actions');
 const { asyncHandler } = require('../utils');
-const models = require('../models');
 const { getById: getCaseById } = require('../case/case-data-access');
 const { getById: getContactById } = require('../contact/contact-data-access');
-const { PostSurvey, Case, Contact } = models;
+const createError = require('http-errors');
```

Comment:
I hate throwing http erros in here, but it's the only way I can modify this without breaking all the 404 tests (they go to 500 if the object is not found or unauthorized if I catch them here).

This had me thinking, should we move the middlewares to the routes files (where we handle http errors)? Rethink the way we are checking the permissions so far? Leave this as it is?

I'm also not super cool with having to do x2 ops to the db when we ask for an update, I feel like that can be improved.

Let me know what you think team.

Reply:
Express middlewares are, like routes in the 'http domain' as far as I'm concerned, so it's normal to throw HTTP errors from them. 

They are a custom extension of the web server for handling HTTP requests, so they are HTTP specific - not like business logic modules that shouldn't care about whether they are being called from HTTP, a CLI, a GUI or w/e

Reply:
Ah that makes sense then! Should we group the middlewares with their respective modules or keep the permissions related middlewares here?